### PR TITLE
REF: Split up parsers into smaller files

### LINKED ISSUE #38930: TST/REF: splitting pandas/io/parsers.py into multiple files
`pandas/io/parsers.py` is close to 4k LOC (3986 on 1.3 master). Would it be reasonable to split it into multiple files (located, say, in a `pandas/io/parsers` directory)?

AFAICT the file contains 4 logical pieces:
- `read_...` methods (~700 LOC)
-  `ParserBase` and related module-level methods ( ~1100 LOC)
- `CParser` and related module-level methods (~400LOC)
- `PythonParser`, its derived classes and related module-level methods (~1800 LOC)

Once #38370 goes in these would be joined by 
- `PyarrowParser` (~100 LOC right now)

IMO this kind of refactor would make the module somewhat easier to grok

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #38930
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

Refactored them into 4 files, similar to described in the issue. If this is accepted I will try to clean this up a bit more through unifying the calls a bit

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
i would update `pandas/io/parsers/__init__.py` to import `read_csv, read_fwf, read_table`, and other public functions here (if any others).

you can leave `.readers` just import them to ```__init__.py``` (then you don't need to change the other modules that import).

otherwise this is a straight move?

**Comment by phofl** [comment]
Thanks, will do that. 

Yes, just moved them around. Will go to refactoring after this

**Comment by phofl** [comment]
I moved the relevant functions up, I changed the import paths for the helper functions in the test functions

**Review by jreback — APPROVED** [review]
lgtm ping on green

**Comment by jreback** [comment]
https://github.com/pandas-dev/pandas/pull/39217/checks?check_run_id=1730796224 looks like a real fail

**Comment by phofl** [comment]
Have to check, this one worked 3 days ago in the pipelines

**Comment by phofl** [comment]
Hm merged master, let's check if this persists. Can not reproduce this locally

**Comment by phofl** [comment]
Additionally this worked the commit before when I forgot a few imports. At least a place to start pinning if it persists

**Comment by phofl** [comment]
Green now

@twoertwein I think you worked with unclosed handles the last few days/weeks? Could you maybe have a look and see if this is familiar?

**Comment by phofl** [comment]
Failing https://github.com/pandas-dev/pandas/runs/1731152527?check_suite_focus=true too, unrelated here then

**Comment by twoertwein** [comment]
> @twoertwein I think you worked with unclosed handles the last few days/weeks? Could you maybe have a look and see if this is familiar?

I assume that most of the ssl warnings come from boto3 (I honestly gave up on them for now in #39047). Wasn't a test like that recently xfailed  (reason s3/boto3 being updated)?

**Comment by phofl** [comment]
Yes, but this test really failed not only raised a warning,

**Comment by twoertwein** [comment]
do you mean the  `OSError: [Errno 98] Address already in use` in `test_server_and_default_headers test`? Is it possible that multiple instances of pytest run on the same machine but they like to use the same port?

or `test_warn_if_chunks_have_mismatched_type`. I think I have looked at this (or a similar) test when working on #39047. As far as I remember, there shouldn't be a need for any "ssl-stuff" in this test. The only resource that might not be closed is `StringIO`. Locally (on linux) I can not re-create most of the ResourceWanrings.

**Comment by phofl** [comment]
Sorry this was not clear enough by myself. I was talking about ``test_warn_if_chunks_have_mismatched_type``. Have the same issue, can not reproduce locally.

File: pandas/io/parsers/readers.py [hunk 1]
```diff
@@ -0,0 +1,1339 @@
...
+        `legacy` for the original lower precision pandas converter, and
+        `round_trip` for the round-trip converter.
+
+        .. versionchanged:: 1.2
+    """
+    kwds["engine"] = "python"
+    return TextFileReader(*args, **kwds)
+
+
+def _clean_na_values(na_values, keep_default_na=True):
```
Comment (jreback) [inline pandas/io/parsers/readers.py:1050]: might be worth moving from here down to another module (validation.py?) in a followup, e.g. keep readers.py for the actual readers

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**